frontend/test/jest/tests/ProductListItem.test.tsx

```diff
@@ -0,0 +1,38 @@
+// @ts-nocheck
```

Comment:
Do we will need this?

Reply:
Yeah, unfortunately, we need this to bypass type checking inside the test, otherwise, we get typechecking errors when setting up the mocks functions and objects like these: 
![image](https://user-images.githubusercontent.com/58952979/179090318-debb2e1f-f819-4a45-82b5-31e2f7219be3.png)
![image](https://user-images.githubusercontent.com/58952979/179090363-c80681a8-ece0-4beb-ae16-aebec173349d.png)

Reply:
I'd recommend that instead we add [`@ts-ignore` comments](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-6.html#suppress-errors-in-ts-files-using--ts-ignore-comments) above the troublesome lines, in order to enable us to have type checking on the rest of the code in this file. That is potentially helpful in quickly finding places that need to be changed when refactoring.

Reply:
Sounds good! Added `@ts-ignore` and removed `@ts-ignore` comments in https://github.com/iFixit/react-commerce/pull/438/commits/90165dc4d20cc331a3b6ca50f0470f118b61ec35.